File: backend/app/models/budget.py
from sqlalchemy import Column, DateTime, ForeignKey, Integer, Numeric, String
from sqlalchemy.orm import relationship
from sqlalchemy.sql import func

from app.database.database import Base


class Budget(Base):
    __tablename__ = "budgets"

    id = Column(Integer, primary_key=True, index=True)
    month = Column(Integer, nullable=False, index=True)
    year = Column(Integer, nullable=False, index=True)
    total_budget = Column(Numeric(12, 2), nullable=False)
    currency = Column(String(10), nullable=False, default="INR")
    created_at = Column(
        DateTime(timezone=True),
        server_default=func.now(),
        nullable=False
    )
    updated_at = Column(
        DateTime(timezone=True),
        server_default=func.now(),
        onupdate=func.now(),
        nullable=False
    )
    user_id = Column(Integer, ForeignKey("users.id"), nullable=False, index=True)

    owner = relationship("User", back_populates="budgets")

File: frontend/src/types/budget.ts
export type Budget = {
  id: number
  month: number
  year: number
  total_budget: number
  currency: string
  created_at: string
  updated_at: string
}

export type BudgetPayload = {
  month: number
  year: number
  total_budget: number
  currency: string
}

export type BudgetQueryParams = {
  skip?: number
  limit?: number
  month?: number
  year?: number
  sort_by?: 'created_at' | 'month' | 'year'
  sort_dir?: 'asc' | 'desc'
}
